socorro/cron/jobs/fetch_adi_from_hive.py

```diff
@@ -183,13 +176,99 @@ class FetchADIFromHiveCronApp(BaseCronApp):
 
+    required_config.namespace('primary_destination')
+    required_config.primary_destination.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+        "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.primary_destination.add_option(
+        'database_class',
+        default=ConnectionContext,
+        doc='the class responsible for connecting to Postgres',
+        reference_value_from='resource.postgresql',
+    )
+
+    required_config.namespace('secondary_destination')
+    required_config.secondary_destination.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+        "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.secondary_destination.add_option(
+        'database_class',
+        default=ConnectionContext,
+        doc='the class responsible for connecting to Postgres',
+        reference_value_from='resource.postgresql',
+    )
+
     @staticmethod
     def remove_control_characters(s):
         if isinstance(s, str):
             s = unicode(s, 'utf-8', errors='replace')
         return ''.join(c for c in s if unicodedata.category(c)[0] != "C")
 
-    def run(self, connection, date):
+    def _database_transaction(
+        self,
+        connection,
+        raw_adi_logs_pathname,
+        target_date
+    ):
+        with codecs.open(raw_adi_logs_pathname, 'r', 'utf-8') as f:
+            pgcursor = connection.cursor()
+            pgcursor.copy_from(
+                f,
+                'raw_adi_logs',
+                null='None',
+                columns=[
+                    'report_date',
+                    'product_name',
+                    'product_os_platform',
+                    'product_os_version',
+                    'product_version',
+                    'build',
+                    'build_channel',
+                    'product_guid',
+                    'count'
+                ]
+            )
+            pgcursor.execute(_RAW_ADI_QUERY, (target_date,))
+
+        # for Bug 1159993
+        execute_no_results(connection, _FENNEC38_ADI_CHANNEL_CORRECTION_SQL)
+
+    def run(self, date):
+
+        db_class = self.config.primary_destination.database_class
+        primary_database = db_class(self.config.primary_destination)
+        tx_class = self.config.primary_destination.transaction_executor_class
+        primary_transaction = tx_class(
+            self.config,
+            primary_database,
+        )
+        transactions = [primary_transaction]
+
+        db_class = self.config.secondary_destination.database_class
```

Comment:
You mean to change 
```python
    required_config.secondary_destination.add_option(
        'database_class',
        default=ConnectionContext,
        doc='the class responsible for connecting to Postgres',
        reference_value_from='resource.postgresql',
    )
```
...to...:
```python
    required_config.secondary_destination.add_option(
        'database_class',
        default='',
        doc='the class responsible for connecting to Postgres',
        reference_value_from='resource.postgresql',
    )
```
I'm not convinced. Its confusing what the value should be. And it's something else you also need to configure if you want to enable the second destination. Right now you just need to override the database credentials. Having to also configure the `database_class` would be another thing you have to get right.